rust/src/applayernewmodbus/newmodbus.rs

```diff
@@ -25,6 +25,160 @@ use super::parser;
+pub const MODBUS_FUNC_READFILERECORD: u8 =      0x14;
+pub const MODBUS_FUNC_WRITEFILERECORD: u8 =     0x15;
+pub const MODBUS_FUNC_MASKWRITEREG: u8 =        0x16;
+pub const MODBUS_FUNC_READWRITEMULTREGS: u8 =   0x17;
+pub const MODBUS_FUNC_READFIFOQUEUE: u8 =       0x18;
+pub const MODBUS_FUNC_ENCAPINTTRANS: u8 =       0x2b;
+pub const MODBUS_FUNC_MASK: u8 =                0x7f;
+pub const MODBUS_FUNC_ERRORMASK: u8 =           0x80;
+
+/* Modbus Diagnostic functions: Subfunction Code. */
+pub const MODBUS_SUBFUNC_QUERY_DATA: u16 =           0x00;
+pub const MODBUS_SUBFUNC_RESTART_COM: u16 =          0x01;
+pub const MODBUS_SUBFUNC_DIAG_REGS: u16 =            0x02;
+pub const MODBUS_SUBFUNC_CHANGE_DELIMITER: u16 =     0x03;
+pub const MODBUS_SUBFUNC_LISTEN_MODE: u16 =          0x04;
+pub const MODBUS_SUBFUNC_CLEAR_REGS: u16 =           0x0a;
+pub const MODBUS_SUBFUNC_BUS_MSG_COUNT: u16 =        0x0b;
+pub const MODBUS_SUBFUNC_COM_ERR_COUNT: u16 =        0x0c;
+pub const MODBUS_SUBFUNC_EXCEPT_ERR_COUNT: u16 =     0x0d;
+pub const MODBUS_SUBFUNC_SERVER_MSG_COUNT: u16 =     0x0e;
+pub const MODBUS_SUBFUNC_SERVER_NO_RSP_COUNT: u16 =  0x0f;
+pub const MODBUS_SUBFUNC_SERVER_NAK_COUNT: u16 =     0x10;
+pub const MODBUS_SUBFUNC_SERVER_BUSY_COUNT: u16 =    0x11;
+pub const MODBUS_SUBFUNC_SERVER_CHAR_COUNT: u16 =    0x12;
+pub const MODBUS_SUBFUNC_CLEAR_COUNT: u16 =          0x14;
+
+/* Modbus Encapsulated Interface Transport function: MEI type. */
+pub const MODBUS_MEI_ENCAPINTTRANS_CAN: u8 =   0x0d;
+pub const MODBUS_MEI_ENCAPINTTRANS_READ: u8 =  0x0e;
+
+/* Modbus Exception Codes. */
+pub const MODBUS_ERROR_CODE_ILLEGAL_FUNCTION: u8 =      0x01;
+pub const MODBUS_ERROR_CODE_ILLEGAL_DATA_ADDRESS: u8 =  0x02;
+pub const MODBUS_ERROR_CODE_ILLEGAL_DATA_VALUE: u8 =    0x03;
+pub const MODBUS_ERROR_CODE_SERVER_DEVICE_FAILURE: u8 = 0x04;
+pub const MODBUS_ERROR_CODE_MEMORY_PARITY_ERROR: u8 =   0x08;
+
+/* Macro to convert quantity value (in bit) into count value (in word):
+ count = Ceil(quantity/8) */
+ //TODO this macro
+// #define CEIL(quantity) (((quantity) + 7)>>3)
+
+/* Modbus Default unreplied Modbus requests are considered a flood */
+pub const MODBUS_CONFIG_DEFAULT_REQUEST_FLOOD: u32 = 500;
+
+/* Modbus default stream reassembly depth */
+pub const MODBUS_CONFIG_DEFAULT_STREAM_DEPTH: u32 = 0;
+
+static mut request_flood: u32 = MODBUS_CONFIG_DEFAULT_REQUEST_FLOOD;
+static mut stream_depth: u32 = MODBUS_CONFIG_DEFAULT_STREAM_DEPTH;
+
+#[repr(packed, C)]
+pub struct NewModbusHeader {
+    pub transaction_id: u16,
+    pub protocol_id: u16,
+    pub length: u16,
+    pub unit_id: u8,
+}
+
+impl NewModbusHeader {
+    pub fn new() -> NewModbusHeader {
+        NewModbusHeader {
+            transaction_id: 0,
+            protocol_id: MODBUS_PROTOCOL_VER,
+            length: 0,
+            unit_id: 0,
+        }
+    }
+
+    //TODO will the simplest parser come here? Will I need a simpler method that will be struct independent? most likely...
```

Comment:
nom parsers should go in `parser.rs`. Here, you can have a function to make a call to that parser.
You'll have to write a little more code to  understand where would it be appropriate to call the parser. (e.g. A lot of times, you'd need to call the parser (from `parser.rs`) in State struct's implementation to access the results easily)